src/engine/StringMapping.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright 2025, University of Freiburg
+// Chair of Algorithms and Data Structures
+// Authors: Johannes Kalmbach <kalmbach@cs.uni-freiburg.de>
+//          Robin Textor-Falconi <textorr@cs.uni-freiburg.de>
+
+#include "engine/StringMapping.h"
+#include "engine/ExportQueryExecutionTrees.h"
+
+#include "index/Index.h"
+
+namespace qlever::binary_export {
+
+// _____________________________________________________________________________
+std::vector<std::string> StringMapping::flush(const Index& index) {
+  LocalVocab dummy;
+  numProcessedRows_ = 0;
+  std::vector<std::string> sortedStrings;
+  sortedStrings.resize(stringMapping_.size());
+  for (auto& [oldId, newId] : stringMapping_) {
+    auto literalOrIri =
+        ExportQueryExecutionTrees::idToLiteralOrIri(index, oldId, dummy, true);
+    AD_CORRECTNESS_CHECK(literalOrIri.has_value());
+    sortedStrings[newId] =
+        std::move(literalOrIri.value().toStringRepresentation());
+  }
+  stringMapping_.clear();
+  return sortedStrings;
+}
+
+// _____________________________________________________________________________
+Id StringMapping::remapId(Id id) {
+  static constexpr std::array allowedDatatypes{
+    Datatype::VocabIndex, Datatype::LocalVocabIndex,
+    Datatype::TextRecordIndex, Datatype::WordVocabIndex};
+  AD_EXPENSIVE_CHECK(ad_utility::contains(allowedDatatypes, id.getDatatype()));
```

Comment:
Also check (can be static_assert), that each datatype is either trivial, or allowed here, or equal to blank node.